### PR TITLE
Enhancement/4052 trigger remote survey

### PR DESCRIPTION
**Pull request by eugene-manuilov**
## Summary

Addresses issue #4052

## Relevant technical choices

* Decided not to create a new hook because there is no sense in it since we would use it only for one component.

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [ ] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/modules/idea-hub/datastore/module-data.js [hunk 1]
```diff
@@ -65,18 +66,48 @@ export const actions = {
...
 				ideaHubData: {
 					lastIdeaPostUpdatedAt,
+					interactionCount,
+				},
+			};
+		}
+		case INCREMENT_INTERACTION_COUNT: {
+			const { ideaHubData = {} } = state;
+
+			let interactionCount = parseInt( ideaHubData.interactionCount );
```
Comment (aaemnnosttv) [inline assets/js/modules/idea-hub/datastore/module-data.js:101]:
```suggestion
			let interactionCount = parseInt( ideaHubData.interactionCount, 10 );
```

File: assets/js/modules/idea-hub/datastore/module-data.js [hunk 2]
```diff
@@ -65,18 +66,48 @@ export const actions = {
...
+			if ( Number.isNaN( interactionCount ) ) {
+				interactionCount = 0;
+			}
```
Comment (aaemnnosttv) [inline assets/js/modules/idea-hub/datastore/module-data.js:104]: We could use `0` as a fallback instead, e.g. `parseInt(...) || 0`?

File: assets/js/modules/idea-hub/datastore/module-data.test.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+	createTestRegistry,
+	unsubscribeFromAll,
+	untilResolved,
+} from '../../../../../tests/js/utils';
+import { enabledFeatures } from '../../../features';
+import { MODULES_IDEA_HUB } from './constants';
+
+describe( 'modules/idea-hub module-data', () => {
+	const baseInfoVar = '_googlesitekitIdeaHub';
+	const baseInfo = {
```
Comment (aaemnnosttv) [inline assets/js/modules/idea-hub/datastore/module-data.test.js:32]: These look like they were copied from the `core/site` tests. Let's rename these to be more specific to this test.
```suggestion
	const ideaHubDataGlobal = '_googlesitekitIdeaHub';
```
As for `baseInfo` here, it looks like it's only used in the one test for selectors, so how about we inline it there instead?

File: includes/Modules/Idea_Hub.php [hunk 4]
```diff
@@ -106,20 +107,23 @@ final class Idea_Hub extends Module
...
 	 *
+	 * @since n.e.x.t
 	 * @var Post_Idea_Text
 	 */
 	private $post_text_setting;
 
 	/**
 	 * Post_Idea_Topics instance.
 	 *
+	 * @since n.e.x.t
```
Comment (aaemnnosttv) [inline includes/Modules/Idea_Hub.php:126]: Nice catch! Let's update these to use the correct version though as `n.e.x.t` would not be accurate.

File: assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/index.js [hunk 5]
```diff
@@ -82,15 +84,19 @@ function DashboardIdeasWidget( props ) {
...
-	const newIdeas = useSelect( ( select ) =>
-		select( MODULES_IDEA_HUB ).getNewIdeas()
+	const [ triggeredSurvey, setTriggeredSurvey ] = useState( false );
+
+	const { newIdeas, savedIdeas, draftIdeas, interactionsCount } = useSelect(
+		( select ) => ( {
+			newIdeas: select( MODULES_IDEA_HUB ).getNewIdeas(),
+			savedIdeas: select( MODULES_IDEA_HUB ).getSavedIdeas(),
+			draftIdeas: select( MODULES_IDEA_HUB ).getDraftPostIdeas(),
+			interactionsCount: select( MODULES_IDEA_HUB ).getInteractionCount(),
```
Comment (aaemnnosttv) [inline assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/index.js:94]: Let's make this singular like it is in the selector (and PHP).

```suggestion
			interactionCount: select( MODULES_IDEA_HUB ).getInteractionCount(),
```

File: assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/index.js [hunk 6]
```diff
@@ -82,15 +84,19 @@ function DashboardIdeasWidget( props ) {
 	const { defaultActiveTabIndex, Widget, WidgetReportError } = props;
 
 	const [ trackedWidgetView, setTrackedWidgetView ] = useState( false );
-
-	const newIdeas = useSelect( ( select ) =>
-		select( MODULES_IDEA_HUB ).getNewIdeas()
+	const [ triggeredSurvey, setTriggeredSurvey ] = useState( false );
+
+	const { newIdeas, savedIdeas, draftIdeas, interactionsCount } = useSelect(
```
Comment (aaemnnosttv) [inline assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/index.js:89]: This also pending the result of our call later.

File: assets/js/components/ModulesList.js [hunk 7]
```diff
@@ -110,5 +110,5 @@ export default function ModulesList( { moduleSlugs } ) {
 }
 
 ModulesList.propTypes = {
-	moduleSlug: PropTypes.arrayOf( PropTypes.string ).isRequired,
+	moduleSlugs: PropTypes.arrayOf( PropTypes.string ).isRequired,
```
Comment (aaemnnosttv) [inline assets/js/components/ModulesList.js:113]: This just fixes an unrelated prop type error, correct?

**Review by aaemnnosttv — COMMENTED** [review]
Thanks @eugene-manuilov ! This looks solid to me, just a few minor things to touch up and then this should be good to go 👍

**Review by felixarntz — CHANGES_REQUESTED** [review]
@eugene-manuilov I found one problem with the implementation while testing.

Regarding your question: In order to get a survey to be returned, you need to technically be on Site Kit version 1.42.0 or higher. In order to do that (and have the proxy recognize it), you'll need to e.g. manually update the version numbers in your Site Kit codebase, and then either wait for the refresh token call to trigger or just disconnect/reset and reconnect. Any of those will update the proxy data with the current Site Kit version used on the site.

**Comment by eugene-manuilov** [comment]
@felixarntz @aaemnnosttv going to close this PR because it contains unwanted changes from the develop branch. Please, take a look at #4100. It contains all changes and addresses code review feedback added for this PR.

**Pull request closed without merging**